main.go

```diff
@@ -425,30 +296,624 @@ func initDaemon(config *menderConfig, dev installer.DualRootfsDevice, env instal
 	return daemon, nil
 }
 
-func doMain(args []string) error {
-	runOptions, err := argsParse(args)
+func strInSlice(str string, slice []string) bool {
+	for _, s := range slice {
+		if str == s {
+			return true
+		}
+	}
+	return false
+}
+
+func filterArguments(args []string) []string {
+	i := 0
+	for {
+		if strInSlice(args[i], deprecatedCommandArgs) { // remove hyphen
+			args[i] = args[i][1:]
+		} else if args[i] == "-info" { // delete and insert
+			args = append(args[:i-1],
+				append([]string{"--log-level", "info"},
+					args[i+1:]))
+		} else if args[i] == "-debug" { // delete and insert
+			args = append(args[:i-1],
+				append([]string{"--log-level", "debug"},
+					args[i+1:]))
+		} else if strInSlice(arg, deprecatedFlags) {
+			args[i] = "-" + args[i] // add an additional hyphen
+		}
+		i++
+		if i == len(args) {
+			break
+		}
+	}
+	return args
+}
+
+func (opts *setupOptionsType) doSetup(c *cli.Context, cfg menderConfig) error {
+	if c.IsSet("help") {
+		cli.ShowAppHelp(c)
+		return nil
+	}
+	if c.GlobalIsSet("config") && !c.IsSet("config") {
+		// Overlapping globalflag
+		// -- command flag takes precedence over global flag.
+		opts.configPath = c.GlobalString("config")
+	}
+	stdin := bufio.NewReader(os.Stdin)
+
+	state := "deviceType"
+
+	// Regex filters
+	validDevice_re, err := regexp.Compile("^[A-Za-z0-9-_]+$")
```

Comment:
Style: avoid underscores in GO. This should be `validDeviceRE` or `validDeviceRegex`